Fix broken previous and next links

codesections opened the pull request:
The existing `previous` and `next` links seem to point in the wrong directions.  They are `next` and `previous` in a blog order (i.e., chronology), but the sidebar menu presents the links in descending order.  This could also be fixed by changing the order in the sidebar
menu, but this fix seems more straightforward.

To see a demo of the existing, broken functionality, see [this demo site](https://book--gutenberg-theme-demo.netlify.com/common_programming_concepts/how-functions-work/).  Pressing the `previous` arrow from section 4.3 *should* take you to 4.2 but actually takes you to 4.4.  The `next` arrow is similarly broken.

codesections committed:
Change previous and next links

The existing `previous` and `next` links seem to point in the wrong
directions.  They are `next` and `previous` in a blog order (i.e.,
chronology), but the sidebar menu presents the links in descending
order.  This could also be fixed by changing the order in the sidebar
menu, but this fix seems more straightforward.

Keats commented on templates/page.html:
```diff
@@ -7,8 +7,8 @@ <h1>{{ page.title }}</h1>
-    {% if page.previous %}
-        <a class="previous" href="{{ page.previous.permalink }}"><</a>
+    {% if page.next %}
+        <a class="previous" href="{{ page.next.permalink }}"><</a>
```
having the next page link have a class or `previous` feels wrong. Next/Previous are correct for sections and the first subsection, can we keep the right semantics by changing the ordering without changing the sidebar?

codesections committed:
Revise templates for correct semantics

This change preserves the semantics of "next" and "previous" links while
also achieving the desired behavior.

It also adds a note to the README explaining the need for `order` and
`weight` variables.

Finally, and slightly unrelated, it adds a note to the README about
redirecting the homepage.

codesections commented:
I updated the pull request with a commit that gets the desired behavior while preserving the semantics of `page.next.permalink` and `page.previous.permalink`.  It does this by reversing the order of the loop, which has an added advantage: it means that the "order" semantics become more intutative.  That is, a page with `order = 1` would be presented as the first within its section, not the last.

In the process of making this change, I discovered that the loop over the `index.subsections` array presents pages in random order if no wieght is set.  I opened an issue about this (Keats/gutenberg#335) but, in the meantime, have added a note in the README instructing the user to set both `weight` and `order`.

I also added a note to the README about the redirect issue from #2, which should close that issue.  I did that here instead of in the seperate pull request to avoid a merge conflict.

codesections commented:
Sounds good.  It looked ok in my testing, so I'm sorry to hear there were issues.  If you'd like, I'd be happy to dig more into this—just let me know what the issue is if you'd like me to keep at it.

Keats commented:
After the discussion about sorting, I've made the smallest fix in #3 I think

Keats commented:
Can you put the README commit in a new PR?
